src/main/java/org/apache/commons/imaging/formats/png/ChunkReader.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.imaging.formats.png;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.imaging.ImageReadException;
+import static org.apache.commons.imaging.common.BinaryFunctions.readAndVerifyBytes;
+import org.apache.commons.imaging.common.bytesource.ByteSource;
+import org.apache.commons.imaging.formats.png.chunks.ChunkLoader;
+import org.apache.commons.imaging.formats.png.chunks.PngChunk;
+import org.apache.commons.imaging.formats.png.chunks.PngChunkIhdr;
+
+// Replaces readChunk in PngImageParser.
+// This should be a public API. This will allow 3rd party apps
+// to read chunks. Shouldn't ever break.
+
+/**
+ * {@code PngReader} objects can be used to read PNG chunks from
+ * an input stream.
+ * 
+ * @author Shukant Pal
+ */
+public class ChunkReader {
+    
+    DataInputStream input;
+    
+    /**
+     * Asserts that the input stream contains the PNG signature at the
+     * very beginning; throws an ImageReadException otherwise.
+     * 
+     * @throws IOException
+     * @throws ImageReadException 
```

Comment:
This javadocs need explanation for each exception to prevent failures in newer JDK versions.